server/lib/src/idm/oauth2.rs

```diff
@@ -687,6 +762,8 @@ impl<'a> Oauth2ResourceServersWriteTransaction<'a> {
                     prefer_short_username,
                     type_,
                     has_custom_image,
+                    #[cfg(feature = "dev-oauth2-device-flow")]
```

Comment:
Since the cfg feature forces false on this if the feature isn't enabled, you don't need to have a cfg guard on the URL here since it will always be None.